feat: add ProgressBar Component

289Volts opened the pull request:
Added progressBar component

## Description

I have implemented the creation of the progressBar component and also ensured cross-browser compatibility. 

## Related Issue

- Resolves #106 

## Checklist

- [X] I have reviewed the Contribution Guidelines linked above.
- [X] I have tested my changes thoroughly and ensured that all existing tests pass.
- [X] I have provided clear and concise commit messages.
- [ ] I have updated the project's documentation as necessary.

Additional Context

https://github.com/abeg-help/frontend/assets/66239216/5a29d195-bc30-4ef7-ab83-97093062d5d6

289Volts committed:
Feat

Added progreeBar component

ryan-zayne commented on components/primitives/ProgressBar/progress-bar.tsx:
```diff
@@ -0,0 +1,27 @@
+const ProgressBar: FC<ProgressBarProps> = ({ value, className, ...props }) => {
+  return (
+    <progress
+      className={clsx(
```
We now have the cn until available👀

So you can try merging the existing PR handling that, and see if you can incorporate it here instead of raw clsx🤝

babblebey commented on styles/globals.css:
```diff
@@ -26,10 +26,20 @@ body {
     rgb(var(--background-start-rgb));
 }
 
+progress[value]::-webkit-progress-bar {
+  @apply bg-abeg-neutral-80 rounded-[4px];
+}
+progress[value]::-webkit-progress-value {
+  @apply bg-abeg-green rounded-[4px];
+}
+progress[value]::-moz-progress-bar {
+  @apply bg-abeg-green rounded-[4px] h-full;
+}
```
It's would be great if we can stick to utility classes only.

We should only extend to `global.css` in the most minimal cases.

This was the case I guess at first, but you should checkout this discussion, it can help achieve this 😉

https://github.com/tailwindlabs/tailwindcss/discussions/3921

289Volts replied:
Thanks for the link

babblebey commented on components/primitives/ProgressBar/progress-bar.tsx:
```diff
@@ -0,0 +1,27 @@
+import { FC, HTMLProps } from "react";
+
+interface ProgressBarProps {
+  value: string; // Provide the progress value as a number (0-100)
```
Let this be `number` type... It's important as that's the data type that is possible we will be passing into the component.

Meanwhile, you can convert it to `string` at the point of usage in the `progress` value property.

ryan-zayne commented on components/primitives/ProgressBar/progress-bar.tsx:
```diff
@@ -0,0 +1,27 @@
+  return (
+    <progress
+      className={clsx(
+        className,
```
Another thing comes to mind btw, isn't the aim of this classname prop to enable customizability and if possible default style overrides from parent Component?

If so, the pr #105 addresses how this would best be implemented

babblebey commented on components/primitives/ProgressBar/progress-bar.tsx:
```diff
@@ -0,0 +1,27 @@
+        "appearance-none bg-abeg-neutral-80 h-2 rounded-[4px] w-full border-none mx-9",
+      )}
+      id="target"
+      value={value}
```
...convert from number to string at this point yea, like so..

```js
value={string(value)}
```

ryan-zayne replied:
Great catch, although there's now a native HTML way of avoiding to do this, but we can get into those technicalities later🥂

babblebey replied:
Great, I'd like to learn about this later for sure. 😉🤘🏾

Ryan-Zayne commented on styles/globals.css:
```diff
@@ -26,10 +26,20 @@ body {
     rgb(var(--background-start-rgb));
 }
 
+progress[value]::-webkit-progress-bar {
```
E remain make I learn all these accessibility kini sha, una don too oppress😑

Nice work here man🤝🤝

babblebey reviewed (changes requested):
Great stuff @289Volts 

Just this few improvement to consider 😉

Plus, kindly include a screenshot of the component in the PR description. For visuals 😉

289Volts committed:
Merge branch 'dev' of https://github.com/abeg-help/frontend into #106-289volts-Feat-Add-ProgressBar-Component

ryan-zayne commented on styles/globals.css:
```diff
@@ -26,10 +26,20 @@ body {
     rgb(var(--background-start-rgb));
 }
 
+progress[value]::-webkit-progress-bar {
```
Loving the effort put into accessibility so far tbh. You doing the lord's work man🥂✨

But I feel these can still somehow be made into utility classes or better still utility plugins at worst instead of resorting to @apply, no?🤔

289Volts replied:
no ko, yes ni

289Volts committed:
Feat

Changed styles in the global files to utility classes with tailwind plugin. Change progres value type to number

ryan-zayne reviewed (approved):
Great work, agba dev🚀

babblebey reviewed (approved):
LG 🤲🏾

Kindly take note of the editted PR title @289Volts.
